### PR TITLE
added search by category functionality

### PR DESCRIPTION
**Pull request by parasnegi2002**
Targets #{2185}

### Description
added the functionality "sort by category "

### CODE REVIEW

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 1]
```diff
@@ -24,6 +24,22 @@ class="form-control" id="search_input" placeholder="search all books"
             v-model="searchKey">
             <button class="btn btn-info ml-2" @click="searchBooks()">Search</button>
         </div>
+        
+        <div class="col-lg-4 col-md-5 col-sm-6 col-xs-12 mr-2 mb-2 p-2 d-flex justify-content-center align-items-center">
+        <span>sort by category</span>
+            <select class="form-control bg-white form-control-st" name="category_name" v-model="searchKeys">
+				<option value="Select Category"></option>
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/index.blade.php:31]: Remove this if its not needed

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 2]
```diff
@@ -24,6 +24,22 @@ class="form-control" id="search_input" placeholder="search all books"
             v-model="searchKey">
             <button class="btn btn-info ml-2" @click="searchBooks()">Search</button>
         </div>
+        
+        <div class="col-lg-4 col-md-5 col-sm-6 col-xs-12 mr-2 mb-2 p-2 d-flex justify-content-center align-items-center">
+        <span>sort by category</span>
```
Comment (gautam071) [inline resources/views/knowledgecafe/library/books/index.blade.php:29]:
```suggestion
        <span>Sort book by category</span>
```

File: app/Http/Controllers/KnowledgeCafe/Library/BookController.php [hunk 3]
```diff
@@ -23,11 +23,12 @@ public function __construct()
...
      * @return \Illuminate\View\View
      */
-    public function index()
+    public function index(Request $request)
     {
         $this->authorize('list', Book::class);
-        $searchString = (request()->has('search')) ? request()->input('search') : false;
-        $books = Book::getList($searchString);
+        $searchCategory = $request->category_name ?? false;
+        $searchString = $request->search ?? false;
```
Comment (gautam071) [inline app/Http/Controllers/KnowledgeCafe/Library/BookController.php:30]: What was the reason for changing the search string variable contnent

**Review by gautam071 — CHANGES_REQUESTED** [review]
Few suggestions

File: app/Http/Controllers/KnowledgeCafe/Library/BookController.php (hunk 3 above)
Comment (parasnegi2002) [reply app/Http/Controllers/KnowledgeCafe/Library/BookController.php:30]: both are same . Its a shortform of this $searchString

**Review by gautam071 — APPROVED** [review]
loooks good

**Pull request merged**